elixir/apps/web/lib/web/live/settings/dns.ex

```diff
@@ -1,7 +1,113 @@
 defmodule Web.Settings.DNS do
   use Web, :live_view
 
+  alias Domain.Config
+
+  defp remove_errors(changeset, field, message) do
+    errors =
+      changeset.errors
+      |> Enum.filter(fn err ->
+        case err do
+          {^field, {msg, _}} ->
+            if msg == message, do: false, else: true
+
+          {_, _} ->
+            true
+        end
+      end)
+
+    %{changeset | errors: errors}
+  end
+
+  defp filter_errors(changeset) do
+    filtered_cs =
+      changeset
+      |> remove_errors(:clients_upstream_dns, "address can't be blank")
+
+    filtered_dns_cs =
+      filtered_cs.changes.clients_upstream_dns
+      |> Enum.map(fn changeset ->
+        case changeset.errors do
+          [] -> changeset
+          _ -> remove_errors(changeset, :address, "can't be blank")
+        end
```

Comment:
You don't need to optimize it here - `remove_errors/3` will behave appropriately for empty lists, but if we do want to exit early for them - we can just add a second function clause: `def remove_errors(%{errors: []} = changeset, ...), do: changeset`:
```suggestion
        remove_errors(changeset, :address, "can't be blank")
```

Reply:
Gotcha.  Yeah, I wasn't sure if this would be unnecessary overhead by calling `remove_errors` on already empty lists, so I did this.  More than happy to remove the `case` statement though 👍